subsys/disk/disk_access.c

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (c) 2018 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+#include <zephyr/types.h>
+#include <misc/__assert.h>
+#include <misc/util.h>
+#include <init.h>
+#include <disk_access.h>
+#include <errno.h>
+#include <device.h>
+
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_DISK_LEVEL
+#include <logging/sys_log.h>
+
+/* list of mounted file systems */
+static sys_dlist_t disk_access_list;
```

Comment:
Please use SYS_DLIST_STATIC_INIT

Reply:
I prefer dynamic as I want to start from disk_access_init I do not want to initialize the before hand.
do you see any issue in not using static init?